core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksCheckpointed.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metastore.rocks;
+
+import alluxio.exception.runtime.AlluxioRuntimeException;
+import alluxio.grpc.ErrorType;
+import alluxio.master.journal.checkpoint.CheckpointInputStream;
+import alluxio.master.journal.checkpoint.Checkpointed;
+
+import io.grpc.Status;
+import org.rocksdb.RocksDBException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Provides default implementations for checkpointing RocksDB databases.
+ */
+public interface RocksCheckpointed extends Checkpointed {
```

Comment:
This is not taking the lock when checkpointint/overwriting the rocksDB. I have a fix out for that https://github.com/Alluxio/alluxio/pull/17060
My fix is still pending on some benchmarking and optimizations, and i estimate it'll be merged in 1-2 weeks. If your change is closer to the end then there's no need to worry about locks. Otherwise if my change gets in first then I'll leave further comments on this PR to add the locking part.